src/Main.java

```diff
@@ -0,0 +1,46 @@
+import java.util.Iterator;
+import java.util.Set;
+import java.util.TreeSet;
+
+/*
+    Task: Create different comparators for the class, add objects in treeset
+    and modify fields while iterating.
+ */
+public class Main {
+
+    public static void main(String[] args) {
+        Main main = new Main();
```

Comment:
add comments when and why exceptions happen

Reply:
Added.